### PR TITLE
docs: explain common usage for injection into request instances

### PR DESCRIPTION
**Pull request by Mazuh**
This is a very common practice that new users have to guess (specially on TypeScript is
not obvious how to do it at a first sight), so I'm improving the docs as suggested by @mcollina
and to meant to close https://github.com/fastify/help/issues/551 on the help repo.

#### Checklist

- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Eomm** [comment]
Ci is failing due markdown linting, could you check it?

**Comment by Mazuh** [comment]
@Eomm I didn't know about .md constraints, sorry. I just fixed it and running `./node_modules/.bin/markdownlint-cli2` here locally now returns 0 (no errors). 🚀

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Eomm — APPROVED** [review]

File: docs/Reference/Hooks.md [hunk 1]
```diff
@@ -649,6 +650,57 @@ fastify.route({
...
+
+A very common use case is, for example, checking user authentication based
+on their token and then storing their recovered data into
+the [Request](./Request.md) instance. This way, your controllers can read it
+easily with `request.authenticatedUser` or whatever you want to call it.
+That's how it might look like:
+
+```js
+fastify.addHook('preParsing', async (request) => {
+  request.authenticatedUser = {
```
[the remark this answers is not in the document]
Comment (Mazuh) [reply docs/Reference/Hooks.md:667]: hm... it's not a requirement, it works without it, the documentation says that's an optimization.

at the end of this recently added text, there are links for plugins and typescript pages for more complex/detailed examples and they include it there.

do you think we need to be more explicit here too about it?

[the remark this answers is not in the document]
Comment (simoneb) [reply docs/Reference/Hooks.md:667]: I was just curious really

**Pull request merged**